tools/build_defs/framework.bzl

```diff
@@ -0,0 +1,284 @@
+load("@bazel_skylib//lib:collections.bzl", "collections")
+load("//tools/build_defs:cc_import.bzl", "create_linking_info")
+
+CC_EXTERNAL_RULE_ATTRIBUTES = dict({
```

Comment:
Add documentation. Do all rules that call cc_external_rule_impl need to have all these attributes?